CONTRIBUTING.md

```diff
@@ -0,0 +1,58 @@
+2. Separate out your changes to `swarm` and changes to `experiments` as much as possible. `swarm` as a point of common use requires more stability and thus will have a more conservative approach to PR - it should be necessary or a significant Qol change rather than a whatever you feel like.
+3. Experiments should work on their own branch and merge into master occasionally. My preferred choice for this is to do this via a PR - it'll do the merging cleanly for you. This does mean you need to reset your remote branch forcibly after a merge though, but I think it'll be a good way to ensure that what's in master is always working correctly and keeps main history clean.
+4. Use your name in the experiment branch i.e. `varun/dropout` or `ben/activations` so it doesn't get deleted by accident. Core changes should be prefixed with swarm `swarm/refactorregime`  as this will trigger tests on push.
+
+## Setting up Python
+
+Swarm needs to be on your `PYTHONPATH` to be importable. I'd suggest using PyCharm since it does the paths for you automatically. However if you'd like to be a bit more agnostic, try [`direnv`](https://direnv.net/) for this. `brew install direnv` + and follow instructions to get it working.
+
+And copy this into `.envrc ` in the project.
+
+```bash
+export PYTHONPATH="$PWD:$PYTHONPATH"
+# if you use a venv do this too
+# source venv/bin/activate
+```
+(You will need to run `direnv allow` for it to work - you should see an error otherwise)
+This will execute every time you `cd` into the project and no matter where you're executing code in that directory it will work. Additionally, it will undo itself when in another directory which is pretty great, especially if you're using a virtualenv. 
+
+## Running an Experiment with Swarm
+
+Swarm primarily provides a core that runs experiments with the same seed for reproducibility and provides an easy way of recording neural net results without need to write a ton of code. 
+
+See the [simple_experiment](experiments/sample/simple_experiment.py) to get a sense of how to use it and [intermediate_experiment](experiments/sample/intermediate_experiment.py) for another example that generates a Hive (group of swarms) across various learning rates.
+
+The basic idea is this:
+
+1. Write a bee_trainer - a function that trains a neural net from start to finish and yields data of interest each epoch. 
+2. If it takes args, use a lambda, or `regimes.make_bee` (which is basically the same thing) to make the `bee_trainer` a function that takes no args.
+3. Use `swarm.core.swarm_train` to generate the results from the experiments.
+4. Get a dictionary back which has each data collected as a `np.array` with the dims represented (bee, epoch, \*data)
+5. Analyse the data or generate an animation from it!
```

Comment:
Ok, this is gooooood!  Except... the hive infrastructure (which I did have running before) isn't included yet... I'll have a fiddle this morning.